libs/island-ui/core/src/lib/Checkbox/Checkbox.tsx

```diff
@@ -16,12 +16,15 @@ export interface CheckboxProps {
   hasError?: boolean
   errorMessage?: string
   value?: string
-  large?: boolean
   filled?: boolean
+  large?: boolean
+  /** subLabel can only be used if the 'large' prop set to true */
+  subLabel?: string
```

Comment:
You might want [to type guard](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIgApQPYAcDOABQN4BQyyAbnADYCuEA-AFzL5hSgDmA3OcjThQuTVgCNs2GhDghSAX1KlQkWIhQBlOmIAycMRBpY8RCAA9IIACb40mHAWRkKg4RHGTpsvhXza9BjQsbBzcfIqkYACeuCjoACoxKAC8dsaOAD7IABTo6UQAZMhauvqG+QCUSjB0IAhgwNggyAAWcPglAYbZ2GIAVqx5DrZZnWVGwxWsvX3IwLZjgfkk-FAQYHRQzT39yO3F-uOVAHR+pYHIAITJqbVWEDCgEFYKSghN7MgwzanZuMODewmZCjQ5LSbIZIAPhWFGAMBybQ6YO6-xMFQqsIoyDRBGO1HoEH4FFx+GOrhExJxw1OKJo-AiJJpBIY-FJ5KElKZJlp50MyAA9ALkAARbAQfDgZDmYBgAA0yAMCDgdHwKGwCLALRQ0ViyC4dCEVmQCG1CAA1q9qiBss4qLQGKwAOT4bAAW3WLW4ToUVVI31t-Ap7mQHAYct91sDLk5IfgNDVEd8dOdgkCToj8j9AbtZy6NFT4wzvqAA) this props so you have better check over the usage of the props

Reply:
Yes definitely! 💯

Reply:
We are going to revisit this at a later date.

Reply:
This is more of a gerneral question and not todo in this PR. 

But when do we decide to break components into smaller components and this for example would be: `CheckBox` and `LargeCheckbox` ? 

I am mainly wondering because of this logic and the styles below are dependent on the large variable currently.

Reply:
I think we are at that point now where we should reconcider how both the `<Checkbox>` and the `<RadioButton>` components are set up.

Reply:
I agree, but out of scope for this PR. Was mainly wondering what the process for it was. Could we pick it up if we feel there is a need for it or is it a decision of the codeowner or something else?

Reply:
It's totally up to you! If you feel a change is needed, feel free to implement it 🙏